Firmware/HomeButtonsArduino/src/network.cpp

```diff
@@ -299,14 +299,23 @@ void Network::_pre_wifi_connect() {
   const auto &static_ip = device_state_.user_preferences().network.static_ip;
   const auto &gateway = device_state_.user_preferences().network.gateway;
   const auto &subnet = device_state_.user_preferences().network.subnet;
+  auto dns = device_state_.user_preferences().network.dns;
+  auto dns2 = device_state_.user_preferences().network.dns2;
   bool ip_ok = static_ip != IPAddress(0, 0, 0, 0);
   bool gw_ok = gateway != IPAddress(0, 0, 0, 0);
   bool sn_ok = subnet != IPAddress(0, 0, 0, 0);
+  if (dns == IPAddress(0, 0, 0, 0)) {
+    dns = gateway;
+  }
+  if (dns2 == IPAddress(0, 0, 0, 0)) {
+    dns2 = IPAddress(1, 1, 1, 1);
+  }
   if (ip_ok && gw_ok && sn_ok) {
-    info("Using static IP %s, Gateway %s, Subnet %s",
+    info("Using static IP %s, Gateway %s, Subnet %s, DNS1 %s, DNS2 %s",
          static_ip.toString().c_str(), gateway.toString().c_str(),
-         subnet.toString().c_str());
-    WiFi.config(static_ip, gateway, subnet);
+         subnet.toString().c_str(), dns.toString().c_str(),
+         dns2.toString().c_str());
+    WiFi.config(static_ip, gateway, subnet, dns, dns2);
```

Comment:
Maybe could avoid dynamic allocation to print logs by using `ip_address_to_static_string` from `utils.h` (`IPAddress::toString()` do allocate memory):
```c++
    info("Using static IP %s, Gateway %s, Subnet %s, DNS1 %s, DNS2 %s",
         ip_address_to_static_string(static_ip).c_str(), ip_address_to_static_string(gateway).c_str(),
         ip_address_to_static_string(subnet).c_str(), ip_address_to_static_string(dns).c_str(),
         ip_address_to_static_string(dns2).c_str());
```